### PR TITLE
support wsearch directives

### PR DESCRIPTION
**Pull request by percyliang**
@kashizui please review

### CODE REVIEW

**Comment by percyliang** [comment]
Fixes #135

File: codalab/lib/worksheet_util.py [hunk 1]
```diff
@@ -571,12 +567,20 @@ def get_schema(args):  # args is a list of schema names
         for arg in args:
             schema += schemas[arg]
         return schema
+    def reset_display_schema():
+        # Reset display to minimize the long distance dependencies of directives
+        if item_type != TYPE_BUNDLE:
+            current_display_ref[0] = default_display
+        # Reset schema to minimize long distance dependencies of directives
+        if item_type != TYPE_DIRECTIVE:
+            current_schema = None
```
Comment (kashizui) [inline codalab/lib/worksheet_util.py]: does this assignment actually modify variable in the outer scope?

File: codalab/lib/worksheet_util.py [hunk 2]
```diff
@@ -774,3 +781,17 @@ def interpret_search(client, worksheet_uuid, data):
...
+def interpret_wsearch(client, data):
+    '''
+    Input: specification of a worksheet search query.
+    Output: worksheet items based on the result of issuing the search query.
+    '''
+    # Get the worksheet uuids
+    worksheet_infos = client.search_worksheets(data['keywords'])
+    items = []
+    for worksheet_info in worksheet_infos:
+        items.append(subworksheet_item(worksheet_info))
```
Comment (kashizui) [inline codalab/lib/worksheet_util.py]: nit: could be a list comprehension?

**Comment by kashizui** [comment]
LGTM besides comment

**Comment by percyliang** [comment]
Good catch, fixed!

**Comment by kashizui** [comment]
LGTM, sorry for merge conflict, I think that was me..

**Pull request merged**